lib/gtfs_realtime_viz.ex

```diff
@@ -36,16 +36,36 @@ defmodule GTFSRealtimeViz do
   Renders the received protobuf files and comments into an HTML fragment that can either
   be opened directly in a browser or embedded within the HTML layout of another app.
   """
-  @spec visualize(term) :: String.t
-  def visualize(group) do
-    [vehicle_archive: vehicles_by_stop_id(group), routes: routes()]
+  @spec visualize(term, %{String.t => [[String.t]]}) :: String.t
+  def visualize(group, opts) do
+    all_locations = locations_we_care_about(opts)
+    vehicles_we_care_about = group
+                             |> State.vehicles
+                             |> vehicles_we_care_about(all_locations)
+
+    [vehicle_archive: vehicles_by_stop_id(vehicles_we_care_about), routes: opts]
     |> gen_html
     |> Phoenix.HTML.safe_to_string
   end
 
-  @spec vehicles_by_stop_id(term) :: [{String.t, %{required(String.t) => [Proto.vehicle_position]}}]
-  defp vehicles_by_stop_id(group) do
-    Enum.map(State.vehicles(group), fn {comment, vehicles} ->
+  def locations_we_care_about(opts) do
+    opts
+    |> Enum.flat_map(fn {_, stop} -> stop end)
+    |> Enum.flat_map(fn stop -> stop end)
+  end
+
+  def vehicles_we_care_about(state, locations_we_care_about) do
+    Enum.map(state,
+      fn {descriptor, position_list} ->
+        filtered_positions = position_list
+        |> Enum.filter(fn position -> position.stop_id in locations_we_care_about end)
```

Comment:
im not thrilled about the way i did this, but i didnt really know a simpler way to do it.  if i missed something obvious, let me know

Reply:
Well, you can make it _terser_ with my friend `update_in` (which, you may remember, [I just love](https://github.com/mbta/realtime-dashboard/pull/16#discussion_r148819121)):

``` ex
def vehicles_we_care_about(state, routes) do
  update_in(state, [Access.all(), Access.elem(1)], fn position_list ->
    Enum.filter(position_list, fn pos -> 
      pos.trip && MapSet.member?(routes, pos.trip.route_id)
    end)
  end)
end
```

Seen here more concretely:

```
iex(6)> s
[{"a", [1, 2, 3, 4]}, {"b", [5, 6, 7, 8]}]
iex(7)> update_in(s, [Access.all(), Access.elem(1)], fn nums -> [0 | nums] end)
[{"a", [0, 1, 2, 3, 4]}, {"b", [0, 5, 6, 7, 8]}]
```

But, that's probably not _simpler_. I like your way just fine, but I do suggest making it a `MapSet` if we're going to be testing if an element is a member of the set a lot, since that will be faster than going through a list each time.